targets/TARGET_NORDIC/TARGET_NRF5/us_ticker.c

```diff
@@ -199,35 +231,23 @@ __STATIC_INLINE void internal_common_rtc_set_interrupt(uint32_t us_timestamp, ui
 
     nrf_rtc_cc_set(COMMON_RTC_INSTANCE, cc_channel, RTC_WRAP(compare_value));
     nrf_rtc_event_enable(COMMON_RTC_INSTANCE, int_mask);
+
+    core_util_critical_section_exit();
 }
 
 void common_rtc_set_interrupt(uint32_t us_timestamp, uint32_t cc_channel,
                               uint32_t int_mask)
 {
     uint32_t prev_overflows;
 
-    while (1)
-    {
+    do {
         prev_overflows = m_common_rtc_overflows;
 
         internal_common_rtc_set_interrupt(us_timestamp, cc_channel, int_mask);
 
         // check in case of preemption by RTC OVF event (apply if call was from a low priority level)
-        if (prev_overflows != m_common_rtc_overflows)
-        {
-            continue;
-        }    // check in case that OVF occurred during execution of a RTC handler (apply if call was from RTC handler)
-        else if (nrf_rtc_event_pending(COMMON_RTC_INSTANCE, NRF_RTC_EVENT_OVERFLOW))
-        {
-            nrf_rtc_event_clear(COMMON_RTC_INSTANCE, NRF_RTC_EVENT_OVERFLOW);
-            // Don't disable this event. It shall occur periodically.
-
-            ++m_common_rtc_overflows;
-            continue;
-        }
-
-        break;
-    }
+        // check in case that OVF occurred during execution of a RTC handler (apply if call was from RTC handler)
+    } while (rtc_ovf_evet_check() != prev_overflows);
```

Comment:
This loop is no longer needed, because now common_rtc_32bit_ticks_get() provides the protection.